lib/audit/catalog_to_moab_existence.rb

```diff
@@ -0,0 +1,30 @@
+require 'druid-tools'
+
+# Catalog to Moab existence check code
+class CatalogToMoabExistence
+
+  # NOTE: this is a bad thing -- HUGE list of objects;  just for now to populate test db
+  # FIXME:  temporarily turning off rubocop
+  # rubocop:disable all
+  def self.whole_db
+    PreservationCopy.all.each do |pc|
```

Comment:
you may want to use `find_each` for this:
http://api.rubyonrails.org/classes/ActiveRecord/Batches.html#method-i-find_each
> Looping through a collection of records from the database (using the ActiveRecord::Scoping::Named::ClassMethods#all method, for example) is very inefficient since it will try to instantiate all the objects at once.
> 
> In that case, batch processing methods allow you to work with the records in batches, thereby greatly reducing memory consumption.
> 
> The find_each method uses find_in_batches with a batch size of 1000 (or as specified by the :batch_size option).

had vague memories of something like that, did some googling, the stack exchange posts suggested `find_each` but weren't very well written, but thankfully the rails docs seem pretty helpful/thorough in this case.

seems like you could just sub `find_each` for `all.each` and keep the contents of the block unchanged, since you still get the object as the block parameter.

Reply:
thanks for this.  I will change this PR.

I'm guessing that in the general case we're going to want to do "the least recently checked X moabs (? on storage_root 5 ?)" or some such.